### PR TITLE
Fix crash when adding a credit card (APPS-2155)

### PR DESCRIPTION
**Pull request by Fabtron**
instead of throwing an exception and crashing the app a loading indicator is now shown until a project is available.

My assumption is that through the custom lifecycle a race condition happens where the fab button can be clicked while no project is set.

APPS-2155

### How to test?
Integrate the sdk into an app where this view is used and apply this patch

```
Index: ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListFragment.kt b/ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListFragment.kt
--- a/ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListFragment.kt	(revision 46053cc1447b8051bfe86f95ce494f4912b5adab)
+++ b/ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListFragment.kt	(date 1739776443474)
@@ -2,12 +2,15 @@
 
 import android.os.Bundle
 import android.view.View
+import androidx.lifecycle.lifecycleScope
 import io.snabble.sdk.Project
 import io.snabble.sdk.Snabble
 import io.snabble.sdk.payment.PaymentCredentials
 import io.snabble.sdk.ui.BaseFragment
 import io.snabble.sdk.ui.R
 import io.snabble.sdk.ui.utils.serializableExtra
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 
 open class PaymentCredentialsListFragment : BaseFragment(
     layoutResId = R.layout.snabble_fragment_payment_credentials_list,
@@ -30,6 +33,9 @@
 
     override fun onActualViewCreated(view: View, savedInstanceState: Bundle?) {
         val v = view as PaymentCredentialsListView
-        type?.let { v.show(it, project) } ?: v.setProject(project)
+        lifecycleScope.launch {
+            delay(3000)
+            type?.let { v.show(it, project) } ?: v.setProject(project)
+        }
     }
 }
```

Switch to the branch and check it out again.

### Definition of Done

- [x] Issue is linked
- [x] All requirements of the issue are fulfilled
- [x] Changelog is updated
- [ ] Documentation is updated
- [x] [Self-Review](https://www.nerdwallet.com/blog/engineering/why-you-should-be-doing-self-reviews/)
- [ ] Review with the Product Owner _(Release-Variante o. Minified Build)_

#### App Tests
- [ ] Minified build has been tested _(aka. Release Build)_
- [ ] Environments have been taken care of _(Production/Staging)_
- [ ] Supported languages have been tested
- [ ] Light-/Dark-Mode has been tested
- [ ] Edge-Cases have been tested
- [ ] Android API Levels have been taken care of _(minSdk?)_

#### Testing
- [ ] Tests have been written _(aka Unit-Tests, Integration-Tests, ...)_

### CODE REVIEW

File: ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListView.java [hunk 1]
```diff
@@ -231,11 +234,15 @@ public void onChanged() {
         }
 
         if (entries.size() == 0) {
-            emptyState.setVisibility(View.VISIBLE);
+            if (project != null) {
+                emptyState.setVisibility(View.VISIBLE);
+                fab.setVisibility(View.VISIBLE);
+            }
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListView.java:240]: Warum nur wenn das project nicht null ist?

File: ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListView.java [hunk 2]
```diff
@@ -69,17 +72,19 @@ public PaymentCredentialsListView(Context context, AttributeSet attrs, int defSt
     private void inflateView() {
         inflate(getContext(), R.layout.snabble_view_payment_credentials_list, this);
         if (isInEditMode()) return;
+        emptyState = findViewById(R.id.empty_state);
+        recyclerView = findViewById(R.id.recycler_view);
+        fab = findViewById(R.id.fab);
+        progressIndicator = findViewById(R.id.circular_indicator);
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListView.java:78]: Inkonsistentes naming

Comment (rekire) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListView.java:78]: Nach dem naming zu Urteilen wird es nie sichtbar gestellt übersehe ich etwas?
